jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryClient.java

```diff
@@ -45,6 +45,29 @@
 
   private static final String PROTOCOL = "https";
 
+  @Nullable private static String userAgentSuffix = "";
+
+  // TODO: Inject via a RegistryClientFactory.
+  /** Sets a suffix to append to {@code User-Agent} headers. */
+  public static void setUserAgentSuffix(@Nullable String userAgentSuffix) {
+    RegistryClient.userAgentSuffix = userAgentSuffix;
+  }
+
+  // TODO: Add more descriptiveness.
+  /** Gets the {@code User-Agent} header to send. */
+  private static String getUserAgent() {
+    String version = RegistryClient.class.getPackage().getImplementationVersion();
+    StringBuilder userAgentBuilder = new StringBuilder();
+    userAgentBuilder.append("jib");
+    if (version != null) {
+      userAgentBuilder.append(" ").append(version);
+    }
+    if (userAgentSuffix != null) {
```

Comment:
so you will append an empty string?

Reply:
Ah, `userAgentSuffix` should have been initialized as `null` above. Fixed.

Reply:
Perhaps a test to cover this?

Reply:
Added test.